src/commit.c

```diff
@@ -476,6 +496,17 @@ int git_commit__parse(void *_commit, git_odb_object *odb_obj)
 	return -1;
 }
 
+int git_commit__parse(void *_commit, git_odb_object *odb_obj)
```

Comment:
Okay, so it seems you created the new function to be able to parse commits from raw data. This looks a lot like what I have been doing in PR #4374, but I went even further and actually decoupled the commits completely from the ODB. Seeing that your approach wouldn't solve my particular use case (as I cannot use the ODB to parse the object), would my approach be able to solve your use case?

Reply:
As I think you saw below, this is all in the name of merging the parsing done when quickly revwalk with the "standard" way. After a quick look, I don't think your approach allows me to do that (it's just decoupling IIUC).

Reply:
Yup, you're right. I tend to review by commits, so I didn't see this until later.